### PR TITLE
[Cleaning3] NULL -> nullptr

### PR DESCRIPTION
**Pull request by mathildemerle**
Will be rebased when https://github.com/medInria/medInria-public/pull/541 is merged.

This makes NULL become nullptr.

:m:

### CODE REVIEW

**Comment by mathildemerle** [comment]
Main commit : https://github.com/medInria/medInria-public/pull/542/commits/7873927f15d11dbeba8b24e0562ed7731471464d

File: src/layers/legacy/medCoreLegacy/parameters/medAbstractParameterL.h [hunk 1]
```diff
@@ -67,107 +66,122 @@ private slots:
...
-    medAbstractIntParameterL(QString name = "unknown int parameter", QObject* parent = 0):
+    medAbstractIntParameterL(QString name = "unknown int parameter", QObject* parent = nullptr):
         medAbstractParameterL(name, parent){}
     virtual int value() const;
+
 public slots:
     virtual void setValue(int value);
     virtual void trigger();
+
+signals:
```
[the remark this answers is not in the document]
Comment (mathildemerle) [reply src/layers/legacy/medCoreLegacy/parameters/medAbstractParameterL.h]: this is the signal block, it's not in the public or protected block. It does not change the code behavior if the signals block is before or after the public/private/protected/etc blocks

**Comment by mathildemerle** [comment]
@Florent2305 beware. This PR is going to be rebased with https://github.com/medInria/medInria-public/pull/541. Do not review twice the code ^^

**Comment by Florent2305** [comment]
Too late lol

**Comment by mathildemerle** [comment]
Geez, it should have been a painful review XD. I'm going to rebase this one now that the other is merged ^^

File: src/layers/legacy/medVtkInria/HWShading/vtkShaderObject.cxx [hunk 2]
```diff
@@ -110,22 +110,18 @@ void vtkShaderObject::Compile()
 
 bool vtkShaderObject::SetGlShaderSource()
 {
-  if (this->SourceText == nullptr)
+    if (this->SourceText == nullptr)
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/HWShading/vtkShaderObject.cxx]: just indentation here, no need to check

File: src/layers/legacy/medVtkInria/vtkImageView/vtkDataSet2DWidget.cxx [hunk 3]
```diff
@@ -169,29 +169,30 @@ void vtkDataSet2DWidgetCommand::Execute(vtkObject *caller,
...
-  
+    if (event == vtkImageView2D::OrientationChangedEvent)
+    {
+        if (this->Widget)
+        {
+            vtkImageView2D *view = vtkImageView2D::SafeDownCast(caller);
+            if (view)
+            {
+                vtkRenderer* renderer = view->GetRenderer();
+                vtkCamera *cam = renderer ? renderer->GetActiveCamera() : nullptr;
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkImageView/vtkDataSet2DWidget.cxx]: nullptr

File: src/layers/legacy/medVtkInria/vtkImageView/vtkImageView.cxx [hunk 4]
```diff
@@ -1234,52 +1239,67 @@ double* vtkImageView::GetBackground() const
...
+    {
         return;
+    }
 
-    vtkCamera *cam = this->GetRenderer() ? this->GetRenderer()->GetActiveCamera() : NULL;
-    if (!cam)
-        return;
-
-    int* extent = this->GetMedVtkImageInfo()->extent;
+    vtkCamera *cam = this->GetRenderer() ? this->GetRenderer()->GetActiveCamera() : nullptr;
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkImageView/vtkImageView.cxx]: nullptr only change with {}

File: src/layers/legacy/medVtkInria/vtkImageView/vtkImageView.cxx [hunk 5]
```diff
@@ -1234,52 +1239,67 @@ double* vtkImageView::GetBackground() const
...
+    }
     if (!this->Get2DDisplayMapperInputAlgorithm() || !this->Get2DDisplayMapperInputAlgorithm()->GetOutputInformation(0))
+    {
         return 1.0;
+    }
 
-    vtkCamera *cam = this->GetRenderer() ? this->GetRenderer()->GetActiveCamera() : NULL;
-    if (!cam)
-        return 1.0;
+    vtkCamera *cam = this->GetRenderer() ? this->GetRenderer()->GetActiveCamera() : nullptr;
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkImageView/vtkImageView.cxx]: nullptr (it's easier if i show them sometimes)

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 6]
```diff
@@ -159,64 +159,64 @@ vtkImageOrientedPointWidget::vtkImageOrientedPointWidget()
 
 vtkImageOrientedPointWidget::~vtkImageOrientedPointWidget()
 {
-  for ( int i = 0; i < this->NumberOfHandles; ++i )
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: indentation of this function and nullptr

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 7]
```diff
@@ -242,95 +242,98 @@ void vtkImageOrientedPointWidget::SetViewProp(vtkProp* prop)
...
 
-    this->CurrentRenderer->AddViewProp(this->LineActor);
-    this->LineActor->SetProperty(this->LineProperty);
-    this->LineActor->PickableOff();
+        this->CurrentRenderer->AddViewProp(this->LineActor);
+        this->LineActor->SetProperty(this->LineProperty);
+        this->LineActor->PickableOff();
 
-    this->InvokeEvent(vtkCommand::EnableEvent,NULL);
+        this->InvokeEvent(vtkCommand::EnableEvent, nullptr);
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: nullptr

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 8]
```diff
@@ -242,95 +242,98 @@ void vtkImageOrientedPointWidget::SetViewProp(vtkProp* prop)
...
+            this->CurrentRenderer->RemoveViewProp(this->Handle[i]);
+        }
 
-    this->CurrentRenderer->RemoveViewProp(this->LineActor);
+        this->CurrentRenderer->RemoveViewProp(this->LineActor);
 
-    this->CurrentHandle = NULL;
-    this->InvokeEvent(vtkCommand::DisableEvent,NULL);
-    this->SetCurrentRenderer(NULL);
+        this->CurrentHandle = nullptr;
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: 3 nullptr

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 9]
```diff
@@ -242,95 +242,98 @@ void vtkImageOrientedPointWidget::SetViewProp(vtkProp* prop)
 
 void vtkImageOrientedPointWidget::SetEnabled(int enabling)
 {
-  if ( !this->Interactor )
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: just indentation. I added a comment on the nullptr line

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 10]
```diff
@@ -668,49 +674,48 @@ void vtkImageOrientedPointWidget::OnLeftButtonDown()
...
-  this->EndInteraction();
-  this->InvokeEvent(vtkCommand::EndInteractionEvent,NULL);
-  this->Interactor->Render();
-  this->CurrentPicker = NULL;
-  
+    this->EventCallbackCommand->SetAbortFlag(1);
+    this->EndInteraction();
+    this->InvokeEvent(vtkCommand::EndInteractionEvent, nullptr);
+    this->Interactor->Render();
+    this->CurrentPicker = nullptr;
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: 2 nullptr

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 11]
```diff
@@ -668,49 +674,48 @@ void vtkImageOrientedPointWidget::OnLeftButtonDown()
...
+    else // continue snap drawing
     {
-    return;
+        return;
     }
 
-  this->State = vtkImageOrientedPointWidget::Start;
-  this->CurrentHandleIndex = this->HighlightHandle(NULL);
+    this->State = vtkImageOrientedPointWidget::Start;
+    this->CurrentHandleIndex = this->HighlightHandle(nullptr);
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: nullptr

File: src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx [hunk 12]
```diff
@@ -723,143 +728,145 @@ void vtkImageOrientedPointWidget::OnMiddleButtonUp()
 
 void vtkImageOrientedPointWidget::OnRightButtonDown()
 {
-  
-  if ( this->State == vtkImageOrientedPointWidget::Snapping ){ return; }
```
Comment (mathildemerle) [inline src/layers/legacy/medVtkInria/vtkWidgetsAddOn/vtkImageOrientedPointWidget.cxx]: function : just indentation and nullptr. No need to check each line

File: src/plugins/legacy/itkDataImage/readers/itkDCMTKDataImageReader.cpp [hunk 13]
```diff
@@ -159,19 +159,25 @@ void itkDCMTKDataImageReaderPrivate::threadDone(itk::DCMTKImageIO::Pointer io)
 
 void itkDCMTKDataImageReaderPrivate::initialiseStatic()
 {
-    if ( ! mutex) {
+    if ( ! mutex)
+    {
         QMutex * m = new QMutex();
-        if ( ! mutex.testAndSetOrdered(NULL, m))
+        if ( ! mutex.testAndSetOrdered(nullptr, m))
```
Comment (mathildemerle) [inline src/plugins/legacy/itkDataImage/readers/itkDCMTKDataImageReader.cpp]: nullptr

File: src/plugins/legacy/medSegmentation/annotations/medAnnIntImageMaskHelper.cpp [hunk 14]
```diff
@@ -35,20 +35,13 @@
 #include <medVtkViewBackend.h>
 
 
-class medAnnIntImageMaskHelperPrivate {
```
Comment (mathildemerle) [inline src/plugins/legacy/medSegmentation/annotations/medAnnIntImageMaskHelper.cpp]: remove empty private class (so, no need of the NULL/nullptr under)

File: src/plugins/legacy/vtkDataMesh/writers/vtkDataManagerWriter.cxx [hunk 15]
```diff
@@ -470,37 +470,38 @@ const char* vtkDataManagerWriter::GetDataSetName()
 //----------------------------------------------------------------------------
 vtkXMLWriter* vtkDataManagerWriter::CreateWriter(vtkMetaDataSet* metadataset)
 {
+    vtkXMLWriter* writer = nullptr;
```
Comment (mathildemerle) [inline src/plugins/legacy/vtkDataMesh/writers/vtkDataManagerWriter.cxx]: only indentation of this function and nullptr here, line 477 and 482

**Comment by mathildemerle** [comment]
@Florent2305 PR updated with this commit https://github.com/medInria/medInria-public/pull/542/commits/f2c3f9adb952ac3f0d373fa08543d3c63c309a9e

**Pull request merged**